test/e2e/infraprovider/api/api.go

```diff
@@ -37,6 +39,14 @@ type Provider interface {
 	GetK8HostPort() uint16 // supported K8 host ports
 }
 
+type Underlay struct {
```

Comment:
should we add info for the kind implementation for Underlay here?  network must be precreated currently for the kind provider if no phsical network name provided.